core/store/src/trie/update.rs

```diff
@@ -15,13 +16,13 @@ use crate::StorageError;
 /// Provides a way to access Storage and record changes with future commit.
 pub struct TrieUpdate {
     pub trie: Arc<Trie>,
-    root: MerkleHash,
```

Comment:
MerkleHash seems to be better since it implies specialization?

Reply:
Yes, if it uses in a proper way.

Everytime I see `MerkleHash` I expect the following things:
1. It's a root of some Merkle Tree we built by our inner tools.
2. It appears because of calling `merklize()`.
3. It's a base to prove with `verify_path()`.

And those ones are important to understand quickly is something provable or not.

Using `MerkleHash` without `merklize()` creates lots of mess. It's become hard to say can something be proven and how easy to prove.

Can we use something neutral there? Like `TrieHash` or `TrieRoot`? I'm also waiting for @mikhailOK #1523 solution because his decisions will define is it Merkle or not.

Reply:
Okay